crates/bevy_ecs/src/query/state.rs

```diff
@@ -78,6 +81,12 @@ where
         }
     }
 
+    /// Takes a query for the given [`World`], checks if the given world is the same as the query, and
+    /// generates new archetypes for the given world.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the `world.id()` is not equal to the query's `world_id`.
```

Comment:
field `world_id` is not public, would it be possible to reformulate to be explicit with the value origin?